core/java8/proxy/src/main/java/org/apache/openwhisk/runtime/java/action/Proxy.java

```diff
@@ -48,7 +51,20 @@ public Proxy(int port) throws IOException {
 
         this.server.createContext("/init", new InitHandler());
         this.server.createContext("/run", new RunHandler());
-        this.server.setExecutor(null); // creates a default executor
+
+        if (Boolean.parseBoolean(System.getenv("__OW_ALLOW_CONCURRENT"))) {
+            ThreadPoolExecutor executor = new ThreadPoolExecutor(
+                    10,                  // Core size.
+                    25,              // Max size.
```

Comment:
I'm not sure if we want to set the max pool size as this becomes a hard bound on the language when the openwhisk setting per function is defined elsewhere. So the user would have to know explicitly that for java they can't increase their concurrency past 25 without getting performance degradation even if they aren't cpu bounded.

Reply:
Yeah. This was basically an arbitrary setting on my part as I was moreso interested in getting something working. We could just as well set the max pool size to something else (or a configurable limit).

Reply:
It would be a much larger change for the system to pass in the configured action limit as a part of the payload on the init request (and not every language would need the value so the system would need special logic per language for this). I think leaving it unbounded for now is probably the easiest way to get the initial working capability and can be optimized later if needed